File: Lista_CSharp/Lista_CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Lista_CSharp
{
    public partial class Form1 : Form
    {
        int c = 0;
        Persona P1 = new Persona();
        String[,] mat = new String[5, 4];
        List<Persona> ListP= new List<Persona>();

        public Form1()
        {
            InitializeComponent();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            mat[c, 0] = txtNombre.Text;
            mat[c, 1] = txtApe.Text;
            mat[c, 2] = txtEdad.Text;
            mat[c, 3] = txtTlf.Text;
            c++;
            txtNombre.Text = "";
            txtApe.Text = "";
            txtEdad.Text = "";
            txtTlf.Text = "";
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            Lista.Items.Clear();
            for (int i = 0; i < c; i++)
            {
                P1.setNombre(mat[i, 0]);
                P1.setApellido(mat[i, 1]);
                P1.setEdad(Int32.Parse(mat[i, 2]));
                P1.setTelefono(mat[i, 3]);
                ListP.Add(P1);
                Lista.Items.Add(ListP.ElementAt(i).getNombre()+" "+ListP.ElementAt(i).getApellido()+" "+ListP.ElementAt(i).getEdad().ToString()+" "+ListP.ElementAt(i).getTelefono());
            }
        }
    }
}

File: Vector_Lista/Vector_Lista/Form1.vb
Public Class Form1
    Dim f As Integer = 0
    '  Dim car As Integer = 0
    Dim mat(5, 3) As String
    Dim P1 As Persona = New Persona()
    Dim list As New List(Of Persona)

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        mat(f, 0) = txtNombre.Text
        mat(f, 1) = txtApe.Text
        mat(f, 2) = txtEdad.Text
        mat(f, 3) = txtTlf.Text
        f += 1
        txtNombre.Text = ""
        txtApe.Text = ""
        txtEdad.Text = ""
        txtTlf.Text = ""
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        ListBox1.Items.Clear()
        For i As Integer = 0 To f - 1
            P1.Pnombre = mat(i, 0)
            P1.Papellido = mat(i, 1)
            P1.Pedad = CInt(mat(i, 2))
            P1.Ptelefono = mat(i, 3)
            list.Add(P1)
            ListBox1.Items.Add(list(i).Pnombre + " " + list.Item(i).Papellido + " " + CStr(list.Item(i).Pedad) + " " + list.Item(i).Ptelefono)
        Next
    End Sub
End Class
